lib/samples/download_preplanned_map_area/download_preplanned_map_area_sample.dart

```diff
@@ -34,130 +32,269 @@ class DownloadPreplannedMapAreaSample extends StatefulWidget {
+            children: [
+              const Spacer(),
+              Text('Select Map', style: Theme.of(context).textTheme.titleLarge),
+              const Spacer(),
+              IconButton(
+                icon: const Icon(Icons.close),
+                onPressed: () => setState(() => _mapSelectionVisible = false),
+              ),
+            ],
+          ),
+          const SizedBox(height: 10),
+          // Create a list tile for the web map.
+          ListTile(
+            title: const Text('Web Map (online)'),
+            trailing: _mapViewController.arcGISMap == _webMap
+                ? const Icon(Icons.check)
+                : null,
+            onTap: () => setMapAndViewpoint(_webMap),
+          ),
+          const SizedBox(height: 20),
+          Text('Preplanned Map Areas:',
+              style: Theme.of(context).textTheme.titleMedium),
+          const SizedBox(height: 20),
+          SizedBox(
+            height: 250,
+            // Create a series of list tiles for the offline map areas.
+            child: ListView.builder(
+                itemCount: _preplannedMapAreas.length,
+                itemBuilder: (context, index) {
+                  return Padding(
+                    padding: const EdgeInsets.symmetric(
+                        vertical: 5.0, horizontal: 0),
+                    child: buildMapAreaListTile(
+                        _preplannedMapAreas.keys.toList()[index]),
+                  );
+                }),
+          ),
+          const SizedBox(height: 20),
+        ],
+      ),
+    );
   }
 
-  Future<ArcGISMap?> _downloadOfflineMap(int preplannedAreaIndex) async {
-    if (_offlineMapTask == null) {
-      return null;
+  // Builds a list tile for each preplanned area and updates the UI depending on the status of the job.
+  Widget buildMapAreaListTile(PreplannedMapArea mapArea) {
+    final title = Text(mapArea.portalItem.title);
+    final job = _preplannedMapAreas[mapArea];
+
+    // If a job has been associated with the preplanned area, check the status and display the relevant UI.
+    if (job != null) {
+      if (job.status == JobStatus.started) {
+        return ListTile(
+          enabled: false,
+          title: title,
+          trailing: Column(
+            children: [
+              // When the job is started, display the progress.
+              CircularProgressIndicator(value: job.progress.toDouble() / 100),
+              Text('${job.progress}%'),
+            ],
+          ),
+        );
+      } else if (job.status == JobStatus.succeeded && job.result != null) {
+        // When the job has succeeded, get the result and then get the map from the result.
+        final map = job.result!.offlineMap;
+        return ListTile(
+          title: title,
+          trailing: _mapViewController.arcGISMap == map
+              ? const Icon(Icons.check)
+              : null,
+          onTap: () => setMapAndViewpoint(map),
```

Comment:
maybe the onTap should be disabled when `_mapViewController.arcGISMap == _webMap`